BrainPortal/app/models/cluster_task.rb

```diff
@@ -1687,6 +1689,151 @@ def docker_commands
+    new_task.share_workdir_with(new_share_wd_tid, "Queued") if new_share_wd_tid # can be nil
+
+    # Sets tool config among tool configs accessible by user of current task
+    new_tool                = new_task.tool
+    accessible_tool_configs = ToolConfig.find_all_accessible_by_user(self.user).where(:tool_id => new_tool.id, :bourreau_id => self.bourreau_id)
+    new_tcid                = accessible_tool_configs.limit(1).raw_first_column(:id)[0] if new_tcid.blank?
+    if new_tcid.blank? || ! accessible_tool_configs.where(:id => new_tcid).exists?
+      raise "Cannot find an acceptable tool config ID for this user, tool, and bourreau.\n"
+    end
+    new_task.tool_config_id = new_tcid
+
+    # Sets task parameters
+    new_task.params = new_params.symbolize_keys
+
+    # Sets task prerequisites
+    new_prereqs.split(",").each do |dep_id|
+      new_task.add_prerequisites_for_setup(dep_id,'Completed')
+    end
+
+    # Sets other task attributes
+    # In the future, we could allow to register results to another data provider
+    # or to submit the task to another bourreau. This would require to carefully
+    # check permissions of the task's current user, though.
+    new_task.description              = new_description
+    new_task.user_id                  = self.user_id
+    new_task.results_data_provider_id = self.results_data_provider_id
+    new_task.bourreau_id              = self.bourreau_id
+
+    # Submits the task
+    new_task.status = "New"
+    new_task.save!
+
+    # Returns subtask id to application as a tiny .cbid file
+    taskid_filename = json_filename.sub(/\.json$/i,".cbid")
+    File.write(taskid_filename, "#{new_task.id}\n" )
+
+    # Add new task as a pre-requisite of the current task if requested
+    if new_task_hash["required-to-post-process"]
+      self.add_prerequisites_for_post_processing(new_task,'Completed')
+      self.save!
+    end
+
+  end
+
+  # Validates a JSON string against
+  # the schema used to define new tasks.
+  def validate_json_string json_string
+    # The JSON schema could be extended with bourreau id and results
+    # data provider id but that would require careful permission
+    # checks. tool-config-id is not mandatory because it's specific
+    # to a CBRAIN installation and cannot be easily obtained by an
+    # external agent. In case no tool config id is provided, the first
+    # accessible tool config id will be selected.
+    schema = {
+      "type"       => "object",
+      "required"   => [ "tool-class", "parameters" ],
+      "properties" => {
+        "tool-class"     => { "type" => "string" },                                         # Class of the new task
+        "tool-config-id" => { "type" => "number" },                                         # Tool config id of the new task
+        "share_wd_tid"   => { "type" => "string" },                                         # ID of other task to share workdir
```

Comment:
OK. NaN added that. I didn't know they were interchangeable, or if not then it means the JSON creation code also need to be adjusted.